### DISCUSSION TITLE
How to manage URL of a deleted object?

### DISCUSSION THREAD

**Discussion by marcops85** [supabase/supabase#30383] [Questions] [answered]
hi guys, I have a problem that I don't know how to deal with, in a table I have a column where I store the URLs of some objects in the storage, the problem is that when these objects are deleted the URLs (which no longer exist) remain there. Is there a way to make sure that when the object in storage is deleted the URL of the object in the column is set to NULL?

**Comment by GaryAustin1** [comment, marked as answer]
That is something you have to take care of.  
You could have the code that deletes the file also update the column in the table.

You could have a trigger function on your column that calls the pg_net or http extension to call the storage API to delete file.  Then you update the column to null or delete the row and the file gets deleted.

Some have attempted to put a trigger on storage.objects to manage a table on file operations, but storage SQL operations are not officially documented and could change.

**Reply by marcops85** [reply]
Thanks so much for the reply, I'm new to supabase and I'm not good with functions, is there any documentation to learn some code to use to create the functions I need?

**Reply by GaryAustin1** [reply]
https://github.com/orgs/supabase/discussions/7067  Has some code samples.

And here I show a more complicated system that uses cron to delete files later in time from when they are deleted.  It does show some use of API calls in a function...  It might be more than you want to tackle as a beginner.
https://github.com/GaryAustin1/supa-file-helper

I don't know of any examples of having a delete trigger on storage.objects to delete a column value from your table but it is likely doable with the risk of future Supabase changes impacting it.
